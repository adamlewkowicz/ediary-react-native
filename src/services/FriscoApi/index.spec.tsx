import { friscoApi } from '.';
import friscoResponseMock from './__mocks__/frisco-query.json';
(global as any).fetch = require('node-fetch');

jest.mock('node-fetch');

test('findByQuery - product is normalized correctly', async () => {
  const name = 'apple';
  const fetchMock = fetch;

  (fetchMock as jest.Mock).mockImplementationOnce(async () => ({
    ok: true,
    async json() {
      return friscoResponseMock;
    }
  }));

  const [normalizedProduct] = await friscoApi.findByBarcode(name);

  expect(fetchMock).toHaveBeenCalledTimes(1);
  expect(normalizedProduct).toMatchSnapshot();
});