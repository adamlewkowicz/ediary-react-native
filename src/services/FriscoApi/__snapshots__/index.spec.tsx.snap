// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findByQuery - product is normalized correctly 1`] = `
Object {
  "_id": "9295",
  "barcode": "54490086",
  "brand": "COCA-COLA",
  "description": "Napój gazowany o smaku cola.<br/>Naturalne aromaty. Bez dodatku substancji konserwujących.<br/>Od 1886 Coca-Cola każdego dnia dostarcza radość i orzeźwienie ludziom na całym świecie.<br/>To najpopularniejszy i najchętniej kupowany napój gazowany w historii, jak również najlepiej znany produkt na świecie.<br/>Tajemnica Coca-Coli tkwi w unikalnej recepturze, opierającej się na składnikach pochodzenia naturalnego, która nie zmieniła się od przeszło 125 lat.",
  "images": Array [
    "https://res.cloudinary.com/dj484tw6k/image/upload/v1500521101/bb/9295.png",
  ],
  "ingredients": Array [
    "woda",
    "cukier",
    "dwutlenek węgla",
    "barwnik E 150d",
    "kwas: kwas fosforowy",
    "naturalne aromaty w tym kofeina",
  ],
  "macro": Object {
    "carbs": 21.2,
    "fats": 0,
    "kcal": 42,
    "prots": 0,
  },
  "name": "Napój gazowany (szkło) (cena zawiera 0,40 zł kaucji za butelkę)",
  "portion": 250,
  "portions": Array [],
  "producer": "COCA-COLA",
}
`;
